File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKIX/CE/DistributionPoint.cs
using System; 

//	DistributionPoint ::= SEQUENCE {
//		distributionPoint [0] IMPLICIT DistributionPointName OPTIONAL,
//		reasons           [1] IMPLICIT ReasonFlags OPTIONAL,
//		cRLIssuer         [2] IMPLICIT GeneralNames OPTIONAL 
//	}

namespace Aladdin.ASN1.ISO.PKIX.CE
{
	public class DistributionPoint : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ChoiceCreator<DistributionPointName  >().Factory(), Cast.O, Tag.Context(0)), 
			new ObjectInfo(new ObjectCreator<BitFlags				>().Factory(), Cast.O, Tag.Context(1)), 
			new ObjectInfo(new ObjectCreator<GeneralNames			>().Factory(), Cast.O, Tag.Context(2)), 
		}; 
		// конструктор при раскодировании
		public DistributionPoint(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public DistributionPoint(IEncodable distributionPointName, BitFlags reasons, 
			GeneralNames cRLIssuer) : base(info, distributionPointName, reasons, cRLIssuer) {}

		public IEncodable	DistributionPointName	{ get { return				 this[0]; }} 
		public BitFlags		Reasons					{ get { return (BitFlags	)this[1]; }}
		public GeneralNames	CRLIssuer				{ get { return (GeneralNames)this[2]; }}
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkix/ce/DistributionPoint.java
package aladdin.asn1.iso.pkix.ce;
import aladdin.asn1.*; 
import aladdin.asn1.iso.pkix.*; 
import java.io.*;

//	DistributionPoint ::= SEQUENCE {
//		distributionPoint [0] IMPLICIT DistributionPointName OPTIONAL,
//		reasons           [1] IMPLICIT ReasonFlags OPTIONAL,
//		cRLIssuer         [2] IMPLICIT GeneralNames OPTIONAL 
//	}

public final class DistributionPoint extends Sequence<IEncodable>
{
    private static final long serialVersionUID = -8234928854145354944L;
    
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ChoiceCreator(DistributionPointName  .class).factory(), Cast.O, Tag.context(0)), 
		new ObjectInfo(new ObjectCreator(BitFlags               .class).factory(), Cast.O, Tag.context(1)), 
		new ObjectInfo(new ObjectCreator(GeneralNames			.class).factory(), Cast.O, Tag.context(2)), 
	}; 
	// конструктор при раскодировании
	public DistributionPoint(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public DistributionPoint(IEncodable distributionPointName, BitFlags reasons, 
		GeneralNames cRLIssuer) 
	{
		super(info, distributionPointName, reasons, cRLIssuer); 
	}
	public final IEncodable     distributionPointName() { return			     get(0); } 
	public final BitFlags       reasons				 () { return (BitFlags      )get(1); }
	public final GeneralNames	crlIssuer			 () { return (GeneralNames  )get(2); }
}
